docs/UserGuide.md

```diff
@@ -170,27 +170,42 @@ Format: `setappointment INDEX d/[YYYY-MM-DD]`
 Example:
 - `setappointment 1 d/2024-11-23`
 
-### Locating beneficiaries by name: `find`
+### Searching and filtering clients: `find`
 
-Finds beneficiaries whose names and address contain any of the given keywords.
+Finds and filters clients using a combination of name, address, and tags.
 
-Format: `find n/KEYWORD [MORE_KEYWORDS] a/KEYWORD [MORE_KEYWORDS]`
+Format: `find n/KEYWORD [MORE_KEYWORDS] a/KEYWORD [MORE_KEYWORDS] t/TAG [MORE_TAGS]`
 
+* At least one of name, address, or tag needs to be entered. 
 * The search is case-insensitive. e.g `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Only the name is searched.
-* Partial names will also be matched e.g. `Han` will match `Hans`
-* Beneficiaries matching at least one keyword will be returned (i.e. `OR` search).
+* The order of the keywords does not matter. 
+  * e.g. `n/Hans Bo` will match name `Bo Hans`
+  * `a/Ave Clementi` will match the address`Clementi Ave`
+* Partial names and addresses will also be matched 
+  * e.g. `Han` will match `Hans`
+  * e.g. `Cle` will match `Clementi`
```

Comment:
i think shud mention that for tags is complete string matching (correct me if i m wrong)